### DISCUSSION TITLE
Can metadata from the pod be added to the source file?

### DISCUSSION THREAD

**Discussion by Rentu** [vectordotdev/vector#22098] [Q&A]
Can metadata from the pod be added to the source file? I need to specify which log files to collect in the pod, using the include option, and at the same time, I need the metadata information of the pod for integration with transform. Relying solely on the filename can be very challenging in complex scenarios.

**Comment by jszwedko** [comment]
Hi @Rentu ,

I'm not sure I understand the question. If you are using the `kubernetes_logs` source, then metadata will be added to all lines read from the file. The fields are documented here: https://vector.dev/docs/reference/configuration/sources/kubernetes_logs/#line-fields

**Reply by jszwedko** [reply]
Also, small nit, this post should have been in Q&A, not Ideas. I moved it.
